test/framework/alertmanager.go

```diff
@@ -194,12 +194,17 @@ func (f *Framework) CreateAlertmanagerAndWaitUntilReady(ctx context.Context, ns
 		return nil, errors.Wrap(err, fmt.Sprintf("creating alertmanager %v failed", a.Name))
 	}
 
-	return a, f.WaitForAlertmanagerReady(ctx, ns, a.Name, int(*a.Spec.Replicas), a.Spec.ForceEnableClusterMode)
+	isAMHTTPS := false
+	if a.Spec.Web != nil && a.Spec.Web.TLSConfig != nil {
+		isAMHTTPS = true
+	}
+
+	return a, f.WaitForAlertmanagerReady(ctx, ns, a.Name, int(*a.Spec.Replicas), a.Spec.ForceEnableClusterMode, isAMHTTPS)
 }
 
 // WaitForAlertmanagerReady waits for each individual pod as well as the
 // cluster as a whole to be ready.
-func (f *Framework) WaitForAlertmanagerReady(ctx context.Context, ns, name string, replicas int, forceEnableClusterMode bool) error {
+func (f *Framework) WaitForAlertmanagerReady(ctx context.Context, ns, name string, replicas int, forceEnableClusterMode, isAMHTTPS bool) error {
```

Comment:
It would simplify the function signature to pass the whole Alertmanager object and let `WaitForAlertmanagerReady()` decide whether it needs http or https.

```suggestion
func (f *Framework) WaitForAlertmanagerReady(ctx context.Context, a *monitoringv1.Alertmanager, replicas int) error {
```